### PR TITLE
Removes whitelist from transport and raiden_service.

### PR DESCRIPTION
**Pull request by err508**
Healthcheck and whitelist are similiar, where healtcheck is more restrictive (verifies matrix userid). Currently, client whitelists before initial sync is complete, e.g. whitelists channel partners from closed channels. 

This PR aims to remove whitelist and instead use healtcheck consistently. In order to prevent race conditions during initial sync, _RetryQueue waits for the transport to start and finish all healthchecks, before potential pending messages are sent.

### CODE REVIEW

**Comment by andrevmatos** [comment]
Whitelist was specifically added to enable transport to know before initial sync which nodes it'd be allowed to accept and process events, specifically invites. If you keep only `start_health_check` (which requires transport to be started, and therefore happens later), all invite events which were sent while the node was offline would be [rejected](https://github.com/raiden-network/raiden/blob/34ba00f71b31a4ec6632f6ecb144cc6d70aea2a9/raiden/network/transport/matrix/transport.py#L662-L666), as they weren't added to the address manager prior to transport initial sync.

**Comment by err508** [comment]
@andrevmatos thx for your comment. My current understanding is, that whitelist gets called _before_  the node is synced on the current chain_state. That would whitelist (though not verify) all channel partners we had a channel open during that time. We also do [`_start_transport`](https://github.com/raiden-network/raiden/blob/b61d261fce35c63c4a2c42819f4f44004aa34cdd/raiden/raiden_service.py#L509), where the difference is that we call it _after_ `AlarmTask.first_run()` is done and use `transport.healtcheck` instead of `transport.whitelist`.

> all invite events which were sent while the node was offline would be rejected, as they weren't added to the address manager prior to transport initial sync.

I was wondering if this would happen, but the tests we have seem to pass without it. I started using the transport internal locks in the queue to prevent races between sending pending messages and healthchecks. To me the problem you describe is in general not client specific: 
1. the transport is supposed to be started _after_ chain is synced, 
2. all nodes we're interested in from that _current_ state should be healtchecked 
3. after that is done, we should process invites, messages etc with a current view of what the transport is actually supposed to transport 

So to summarize my understanding: the `whitelist` call whitelists peers from an unsynced chain state, which might be a subset or a superset (i.e. channels were opened or closed while we were offline) of the the nodes we actually want to whitelist in the current chain state. `_start_transport` healthchecks all nodes we're actually want to healtcheck in the current state, so I can't see a reason to use both, especially as whitelist has weaker verification and we should - in general - be economical with peers we whitelist and accept messages from.

**Comment by hackaugusto** [comment]
On an additional note, @andrevmatos could you please provide a state machine diagram of the matrix transport? It is important to document why we have some states and why some of the things must happen in a given order.

**Comment by ulope** [comment]
@err508 hm I thought we had agreed not to do this now. But we were jumping between topics in the last call quite a bit so maybe we didn’t quite settle that topic.

Note to self: more documentation of call outcomes.

> On an additional note, @andrevmatos could you please provide a state machine diagram of the matrix transport? It is important to document why we have some states and why some of the things must happen in a given order.

@hackaugusto that’s one of the things I’m preparing for the workshop on Monday.

**Comment by err508** [comment]
Sorry @ulope I remember we agreed I should discuss this with @andrevmatos, I figured doing a PR would be the most efficient way to do so

**Comment by andrevmatos** [comment]
@err508 

> I was wondering if this would happen, but the tests we have seem to pass without it. I started using the transport internal locks in the queue to prevent races between sending pending messages and healthchecks. To me the problem you describe is in general not client specific:
>
> - the transport is supposed to be started after chain is synced,
> - all nodes we're interested in from that current state should be healtchecked
>  - after that is done, we should process invites, messages etc with a current view of what the transport is actually supposed to transport

I don't think there's tests specific for this case. We would need to go online and then offline, have a partner come online and invite us to a room, and come back online to get this invite on transport initialSync. `start_healthcheck` is called after chain sync, but also after transport initial sync. With the possible partners withelisted/healthchecked only after transport initialSync, any event coming on this initial sync are ignored for these not-yet-whitelisted users. That's not a problem for e.g. messages, as they're retried, but one-shot events like invites are lost forever if they aren't accepted on this initial sync.

**Comment by err508** [comment]
I think you're correct when describing the current state, however:

> and come back online to get this invite on transport initialSync. start_healthcheck is called after chain sync, but also after transport initial sync.

The idea is to block transport initial sync until start_healthcheck is done. start_healthcheck again is called after chain sync, where the set of nodes we actually want to healthcheck is derived from the _current_ chain state, not from a past state. 

Imo the way we do it right now is broken, [especially](https://github.com/raiden-network/raiden/pull/4042/files#diff-063d1686c23ac92538ea0d0df8face3bL485) this part, where we circumvent our own user_id verification in healtcheck, if we already called whitelist (without id verification) before.
